### PR TITLE
:green_book: Update the documentation.

### PR DESCRIPTION
**Pull request by sholsapp**
I've tried to rewrite a lot of our documentation to be more descriptive
in areas that we previously glossed over or hand waved. A lot of recent
issues or discussions have inspired a lot of the content.

Other things:

- Unified our badges/shields under shields.io and made the styling
consistent.
- Split out detailed documents into doc, and reference those, instead of
inlining everything in the top level README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,132 +1,104 @@
...
 
-# About
+[![CircleCI](https://img.shields.io/circleci/project/github/linkedin/pygradle.svg?style=flat-square)]()
+[![license](https://img.shields.io/github/license/linkedin/pygradle.svg?style=flat-square)]()
+[![Bintray](https://img.shields.io/bintray/v/linkedin/maven/pygradle-plugin.svg?style=flat-square)]()
+[![Linkedin](https://img.shields.io/badge/opensource-linkedin-blue.svg?style=flat-square)]()
 
-The PyGradle build system is a set of Gradle plugins that can be used to build Python artifacts. Artifacts produced by
-PyGradle are forward and backward compatible with artifacts produced with Python's setuptools library.
+PyGradle is a enterprise Python build system.
```
Comment (sixninetynine) [inline README.md]: "an"

File: README.md [hunk 2]
```diff
@@ -1,132 +1,104 @@
...
 
-## Why should I use PyGradle?
+PyGradle leverages [Gradle](https://gradle.org/) to empower Python's existing
+ecosystem to solve problems like dependency management, polyglot projects, and
+[lots more](https://engineering.linkedin.com/blog/2016/08/introducing--py-gradle--an-open-source-python-plugin-for-gradle).
+LinkedIn has been using PyGradle for several years to successfully wrangle
```
Comment (sixninetynine) [inline README.md]: maybe say 'manage' instead of wrangle? haha 🐮

File: docs/compatibility.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+PyGradle has basic support for Windows. We would like to note that pex is not
+offically supported on Windows, while it seems to work properly we cannot
+guarantee it.
+
+We're eager to review pull requests from our Windows user base!
+
+### Caveats
+
+- The integration tests require that Python be available in your PATH.
+- The pex format defaults to fat.
```
Comment (sixninetynine) [inline docs/compatibility.md]: maybe add a blurb here about fat pex vs thin pex, as this is our own nomenclature (not terminology used by other pex users)

File: README.md (hunk 1 above)
Comment (sholsapp) [inline README.md]: Fixed thanks

**Comment by ethankhall** [comment]
Thanks!

[ci skip]

**Pull request merged**